book/src/pil/expressions.md

```diff
@@ -35,11 +35,39 @@ Elementary expressions are
 - string literals, written in double quotes, e.g. ``"hello"``
 - array literals written in square brackets, e.g. ``[1, 2, 3]``
 - tuples, having at least two elements, e.g. `(1, "abc")`
+- statement blocks (see below)
 - match expressions (see below).
 - if expressions (see below).
 
 Parentheses are allowed at any point to force precedence.
 
+### Statement Blocks
+
+A ``{``-``}``-delimited block can be used everywhere where an expression is expected.
+
+It has the form ``{ <statement> ; <statement> ; ... ; <expression> }``,
+i.e. a sequence of statements followed by an expression and the value of the statement block
+is the value of the final expression.
+
+The statements can either be expressions or let statements: ``let x = ...`` / ``let x;`
+
+Let statements with value can be used everywhere, they just bind an expression to a local variable
+and allow to avoid repeating the expression.
+
+Let statements without value create a new witness column and are only allowed inside [``constr``-functions](#constr-and-query-functions).
+
+Similarly, an expression at statement level can be used to create new constraints that are added to the global constraint set
+and this can only be done inside a [``constr``-functions](#constr-and-query-functions).
+
+Note that you can always create constraints and return them from a function, even in [pure function](#constr-and-query-functions).
+
+
+Example:
```

Comment:
Is this an example of the `Note...` par above?

Reply:
No, let me move it further up.